### PR TITLE
Add documentation for date variable type

### PR DESCRIPTION
**Pull request by lluzak**
Basic documentation for date variable type introduced in https://github.com/easolhq/easol/pull/9104

### CODE REVIEW

**Comment by lluzak** [comment]
[Introduce a date variable](https://trello.com/c/fjWtKW1T/374-introduce-a-date-variable)

**Review by kylebyrne — COMMENTED** [review]
Looks fine to me but will leave to @paulina-a to give the final ✅

File: docs/theme_architecture/blocks/schema/variables/variable_types/date.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    type: date
+    default: today
+---
+{% assign my_variable_epoch = my_variable | date: "%s" %}
+{% assign current_date = "today" | date: "%s" %}
+
+{% if my_variable_epoch > current_date %}
+  <p>My date variable is after current date</p>
+{% else %}
+    <p>It isn't</p>
```
Comment (paulina-a) [inline docs/theme_architecture/blocks/schema/variables/variable_types/date.md:71]: tiny nit: could the indentations on these be the same please?

**Review by paulina-a — APPROVED** [review]
Thanks @lluzak this is super clear!

**Pull request merged**